### PR TITLE
Update extensibility pattern

### LINKED ISSUE #295: Update extensibility pattern documentation
To account for the changes made in https://github.com/OpenZeppelin/cairo-contracts/pull/240 as discussed in https://github.com/OpenZeppelin/cairo-contracts/discussions/267.

### PR DESCRIPTION
**Pull request by martriay**
Resolves #295. I think the bits about constructors show that we need to differentiate between a contract constructor and a "library constructor". I think this is confusing enough to rollback and call them initializers.

### CODE REVIEW

File: docs/Extensibility.md [hunk 1]
```diff
@@ -29,20 +29,32 @@ To minimize risk, boilerplate, and avoid function naming clashes, we follow thes
...
-* All function and storage variable names must be prefixed with the file name to prevent clashing with other libraries (e.g. `ERC20_approve` in the `ERC20` library)
-* Must not implement any `@external` or `@view` functions
-* Must not implement constructors
-* Must not call initializers on any function
-* Should implement initializer functions to mimic construction logic if needed (as any other library function, never as `@external`)
+Considering four type of functions:
+
+- `A`: internal to a library, not meant to be used outside the module
+- `B`: part of the public API of a library
+- `C`: subset of B that is meant to be exported as-is by contracts
```
Comment (martriay) [inline docs/Extensibility.md:36]: should we call them private, internal, and public? I don't want to coin any term that might conflict with the language down the road, or mix programming concepts and confuse people.

Comment (andrew-fleming) [reply docs/Extensibility.md:36]: I agree that those could definitely work^. I wonder if it'd be easier to use a more literal approach:
`internal`: internal to the lib...
`public`: part of the public API...
`complete`, `finished`, or something like that: ...meant to be exported as-is...

Comment (martriay) [reply docs/Extensibility.md:36]: probably `external`

Comment (koloz193) [reply docs/Extensibility.md:36]: ^+1

Comment (milancermak) [reply docs/Extensibility.md:36]: `external` has a clash with `@external` - a `@view` func is also `external` 

may I suggest `final`?

Comment (martriay) [reply docs/Extensibility.md:36]: but it's not really `final` since they can still be extended, most notably `ERC20.transfer`

Comment (frangio) [reply docs/Extensibility.md]: External makes sense to me. The clash with `@external` seems benign given that external functions should generally be wrapped in an `@external` function, no?

**Review by andrew-fleming — APPROVED** [review]
Looks good to me :)

**Pull request merged**